core/src/test/java/org/infinispan/expiration/impl/ExpirationFunctionalTest.java

```diff
@@ -135,4 +135,16 @@ public void testExpirationMaxIdleInExec() throws Exception {
                (k, ice) -> { throw new RuntimeException("No task should be executed on expired entry"); });
       }
    }
+
+   public void testExpiredEntriesCleared() {
+      for (int i = 0; i < SIZE; i++) {
+         cache.put("key-" + i, "value-" + i,-1, null, i, TimeUnit.MILLISECONDS);
+      }
+
+      // This should expire approximately half of the entries
+      timeService.advance(SIZE / 2);
```

Comment:
Do we really need to test with more than 1 entry?

Reply:
I wanted to test both expired entries and entries that can expire being removed. I can do 2 if you want and it still holds that.